libminifi/src/c2/C2Agent.cpp

```diff
@@ -435,6 +437,27 @@ void C2Agent::handle_c2_server_response(const C2ContentResponse &resp) {
         update_sink_->drainRepositories();
         C2Payload response(Operation::ACKNOWLEDGE, resp.ident, false, true);
         enqueue_c2_response(std::move(response));
+      } else if (resp.name == "corecomponentstate") {
```

Comment:
This has only been written to make sure that it fits the architecture, it hasn't even been tested manually. Once https://github.com/apache/nifi-minifi-cpp/pull/743 is done and merged, I can probably write reasonable tests for it.

Reply:
Let's revisit this after rebasing.

Reply:
This is unfortunately still very much non-trivial to do (unlike testing DESCRIBE, which I do now), so I've added a TODO for it as suggested.